src/lib/output/themes/default/partials/header.tsx

```diff
@@ -5,12 +5,13 @@ import type { PageEvent } from "../../../events";
 import { type Reflection, ReflectionKind } from "../../../../models";
 
 export const header = (context: DefaultThemeRenderContext, props: PageEvent<Reflection>) => {
-    const HeadingLevel = props.model.isProject() ? "h2" : "h1";
+    const opts = context.options.getValue("headings");
+
     return (
         <div class="tsd-page-title">
             {!!props.model.parent && <ul class="tsd-breadcrumb">{context.breadcrumb(props.model)}</ul>}
-            {!props.model.isDocument() && (
-                <HeadingLevel class={classNames({ deprecated: props.model.isDeprecated() })}>
+            {(!props.model.isProject() || opts.project) && (!props.model.isDocument() || opts.document) && (
```

Comment:
Hmmmm.... I've realized that this is actually kind of problematic due to the readme. Both the index and the root modules pages are rendered with the project as the model.... and it's probably somewhat confusing that changing the project option will affect both of the pages... 

Not quite sure how we ought to handle this yet... going to sleep on it.

Reply:
Yea, I agree, not sure how that should be handled. We could add a `headings.readme` flag and then in `header.tsx` use something like the following:

```tsx
//...
{
    (
        !props.model.isProject() ||
        (props.url === "index.html" && opts.readme) ||
        (props.url !== "index.html" && opts.project)
    ) &&
    (!props.model.isDocument() || opts.document) &&
    (<h1...> ... </h1>)
}
// ...
```

Reply:
On second thought... the project page is really just another module, so this check should probably just be for index.html, I think we should always include the title on the modules page.